subscription-manager.spec

```diff
@@ -131,36 +119,34 @@ Source2: subscription-manager-rpmlintrc
 # at the start of a line making meaningful indentation impossible.
 
 Requires:  iproute
-Requires:  %{py_package_prefix}-iniparse
-Requires:  %{py_package_prefix}-decorator
+Requires:  python3-iniparse
+Requires:  python3-decorator
 Requires:  virt-what
-Requires:  %{rhsm_package_name} = %{version}-%{release}
+Requires:  python3-subscription-manager-rhsm = %{version}-%{release}
 Requires: subscription-manager-rhsm-certificates
 %ifarch %{dmidecode_arches}
 Requires: dmidecode
 %endif
 
 %if 0%{?suse_version}
-Requires: %{py_package_prefix}-python-dateutil
-Requires: %{py_package_prefix}-dbus-python
+Requires: python3-python-dateutil
+Requires: python3-dbus-python
 Requires: logrotate
 Requires: cron
-Requires: %{py_package_prefix}-gobject2
+Requires: python3-gobject2
 Requires: libzypp
-Requires: %{py_package_prefix}-zypp-plugin
+Requires: python3-zypp-plugin
 %else
-Requires: %{py_package_prefix}-dateutil
-Requires: %{py_package_prefix}-dbus
+Requires: python3-dateutil
+Requires: python3-dbus
 Requires: python3-gobject-base
 %endif
 
-# rhel 8 has different naming for setuptools going forward
-# on newer rhels and Fedora setuptools is not needed on runtime at all
-%if (0%{?rhel} && 0%{?rhel} == 8)
-Requires:  platform-python-setuptools
-%endif
-
 %if %{use_dnf}
+BuildRequires: cmake
+BuildRequires: gcc
+BuildRequires: json-c-devel
+BuildRequires: libdnf-devel >= 0.22.5
```

Comment:
This does not seem right.

Why should be `cmake`, `gcc`, `json-c-devel` and `libdnf-devel` required to be able to build subscription-manager? These tools (except `gcc`) and libraries are required only by `libdnf-plugin-subscription-manager`. These tools were required when `libdnf-plugin-subscription-manager` was part of another RPM, but it is not true anymore.

Packages `python3-dnf`, `python3-dnf-plugins-core` and `python3-librepo` are required only, when `create_libdnf_rpm` is set. It does not make sense. Are these packages actually required by subscription-manager? Shouldn't be these packages required only by `dnf-plugin-subscription-manager` rpm instead?

Reply:
This block is moved from the same that used to be later on in the spec file.

In particular, inside a `%if %{use_dnf}` block, there were these lines repeated in both the "true" and "false" branches of the `%if %{create_libdnf_rpm}` block inside it.

Please check the message of the 3rd commit, i.e. `spec: deduplicate BuildRequires for use_dnf`.